### PR TITLE
New package(s): py-pydeps and py-stdlib-list

### PR DESCRIPTION
**Pull request by aweits**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pydeps/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    homepage = "https://pypi.python.org/project/pydeps"
+    url      = "https://pypi.io/packages/source/p/pydeps/pydeps-1.7.1.tar.gz"
+
+    version('1.9.0', sha256='ba9b8c7d72cb4dfd3f4dd6b8a250c240d15824850a415fd428f2660ed371361f')
+    version('1.7.1', sha256='7eeb8d0ec2713befe81dd0d15eac540e843b1daae13613df1c572528552d6340')
+
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-enum34', type=('build', 'run'), when='^python@:3.3.99')
+    depends_on('py-stdlib-list', type=('build', 'run'))
+    depends_on('py-pyyaml', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pydeps/package.py:19]: Where do you see this dependency listed?

File: var/spack/repos/builtin/packages/py-stdlib-list/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    docs to get said lists."""
+
+    homepage = "https://pypi.python.org/project/stdlib-list"
+    url      = "https://pypi.io/packages/source/s/stdlib-list/stdlib-list-0.5.0.tar.gz"
+
+    version('0.6.0', sha256='133cc99104f5a4e1604dc88ebb393529bd4c2b99ae7e10d46c0b596f3c67c3f0')
+    version('0.5.0', sha256='5311a95812ebce3c5ad0b1f6ded798ce6f945d157075d166a5426c2da75a6625')
+
+    depends_on('py-functools32', when="^python@:3.1", type=('build', 'run'))
+    depends_on('py-sphinx', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-stdlib-list/package.py:19]: Remove this dep, it's only needed to build the docs

File: var/spack/repos/builtin/packages/py-stdlib-list/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    for Python, along with the code for scraping the official Python
+    docs to get said lists."""
+
+    homepage = "https://pypi.python.org/project/stdlib-list"
+    url      = "https://pypi.io/packages/source/s/stdlib-list/stdlib-list-0.5.0.tar.gz"
+
+    version('0.6.0', sha256='133cc99104f5a4e1604dc88ebb393529bd4c2b99ae7e10d46c0b596f3c67c3f0')
+    version('0.5.0', sha256='5311a95812ebce3c5ad0b1f6ded798ce6f945d157075d166a5426c2da75a6625')
+
+    depends_on('py-functools32', when="^python@:3.1", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-stdlib-list/package.py:18]: Where do you see this dependency listed?

Comment (aweits) [reply var/spack/repos/builtin/packages/py-stdlib-list/package.py:18]: setup.py, "install_requires",
also:
```
coltrane (/tmp/stdlib-list-0.6.0):pydeps --external ./stdlib_list
[
    "functools32", 
    "sphinx"
]
['functools32', 'sphinx']

```

File: var/spack/repos/builtin/packages/py-stdlib-list/package.py (hunk 2 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-stdlib-list/package.py:19]: will do

File: var/spack/repos/builtin/packages/py-stdlib-list/package.py (hunk 3 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-stdlib-list/package.py:18]: Ah, in that case:
```suggestion
    depends_on('py-functools32', when="@0.6.0: ^python@:3.1", type=('build', 'run'))
```
or just remove version 0.5.0.

File: var/spack/repos/builtin/packages/py-pydeps/package.py (hunk 1 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-pydeps/package.py:19]: it's (optionally) imported in pydeps/py2depgraph.py, also in tests (I'll need to add the test dep flag):
```
coltrane (GIT/pydeps):pydeps --external ./pydeps
[
    "configparser", 
    "enum", 
    "stdlib_list", 
    "yaml"
]
['configparser', 'enum', 'stdlib_list', 'yaml']

```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pydeps/package.py:19]: I've never used pydeps before, interesting.

Comment (aweits) [reply var/spack/repos/builtin/packages/py-pydeps/package.py:19]: heh, I only found it after looking for something to do python dep sanity-checking in the context of spack packages specifically.

Comment (aweits) [reply var/spack/repos/builtin/packages/py-pydeps/package.py:19]: Tests are not included in the tarball, only the git repo. So this should be good as-is.

File: var/spack/repos/builtin/packages/py-stdlib-list/package.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class PyStdlibList(PythonPackage):
+    """This package includes lists of all of the standard libraries
+    for Python, along with the code for scraping the official Python
+    docs to get said lists."""
+
+    homepage = "https://pypi.python.org/project/stdlib-list"
+    url      = "https://pypi.io/packages/source/s/stdlib-list/stdlib-list-0.5.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-stdlib-list/package.py:13]:
```suggestion
    url      = "https://pypi.io/packages/source/s/stdlib-list/stdlib-list-0.6.0.tar.gz"
```

File: var/spack/repos/builtin/packages/py-pydeps/package.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+class PyPydeps(PythonPackage):
+    """Python module dependency visualization."""
+
+    homepage = "https://pypi.python.org/project/pydeps"
+    url      = "https://pypi.io/packages/source/p/pydeps/pydeps-1.7.1.tar.gz"
+
+    version('1.9.0', sha256='ba9b8c7d72cb4dfd3f4dd6b8a250c240d15824850a415fd428f2660ed371361f')
+    version('1.7.1', sha256='7eeb8d0ec2713befe81dd0d15eac540e843b1daae13613df1c572528552d6340')
+
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pydeps/package.py:16]: Is this actually used at run-time?

Comment (aweits) [reply var/spack/repos/builtin/packages/py-pydeps/package.py:16]: yes it is, from pydeps itself:
```
from pkg_resources import load_entry_point
```

**Pull request merged**